src/ShareRoutes.jsx

```diff
@@ -32,6 +32,43 @@ function Forward({appPrefix}) {
 }
 
 
+/**
+ * Check if input is a valid url
+ * @param {Object} input
+ * @return {boolean} return true if url is found
+ */
+export function isValidModelURL(input) {
+  {/* eslint-disable-next-line */}
+  const urlRegex = '((http|https)://)(www.)?' + '[a-zA-Z0-9@:%._\\+~#?&//=]{2,256}\\.[a-z]' + '{2,6}\\b([-a-zA-Z0-9@:%._\\+~#?&//=]*)'
+  const isValid = input.match(urlRegex)
+  if (isValid) {
+    return true
+  } else {
+    return false
+  }
+}
+
+
+/**
+ * construct a valid path to the GitHUB model
+ * @param {Object} input
+ * @return {string} model URL
+ */
+export function constructModelURL(input) {
+  const url = new URL(input)
+  const URLParamtersArr = url.pathname.split('/')
+  const URLParameters = {
+    org: URLParamtersArr[1],
+    repo: URLParamtersArr[2],
+    branch: URLParamtersArr[4],
+    fileName: URLParamtersArr[5],
+  }
+  {/* eslint-disable-next-line */}
+  const modelPath = `/share/v/gh/${URLParameters.org}/${URLParameters.repo}/${URLParameters.branch}/${URLParameters.fileName}`
```

Comment:
so this was where I was wondering if moving the function to this class might help.  Can't this string prefix be shared with some of the other code here?

Reply:
I am not sure what you mean, can you please clarify?

Reply:
Sorry, I mean using the installPrefix and appPrefix instead of hardcoding them

Reply:
oh the share and /v/gh ? but they are passed in sharedRoutes which is a separate function